### ISSUE TITLE
error: option --cpp_implementation not recognized

### ISSUE THREAD

**Issue by adamjstewart** [protocolbuffers/protobuf#12390]
**What version of protobuf and what language are you using?**

Version: 4.22.1
Language: Python

**What operating system (Linux, Windows, ...) and version?**

OS: macOS (arm64)
Version: 13.3

**What runtime / compiler are you using (e.g., python version or gcc version)**

Python: 3.10.10
Compiler: Apple Clang 14.0.3

**What did you do?**

```console
$ wget https://files.pythonhosted.org/packages/source/p/protobuf/protobuf-4.22.1.tar.gz
$ tar xzf protobuf-4.22.1.tar.gz 
$ cd protobuf-4.22.1/
$ python setup.py build --cpp_implementation
```

**What did you expect to see**

I expected to build the C++ implementation, as documented in the README. This used to work fine in protobuf 4.21.

**What did you see instead?**

The `--cpp_implementation` flag seems to no longer be valid:
```
usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: setup.py --help [cmd1 cmd2 ...]
   or: setup.py --help-commands
   or: setup.py cmd --help

error: option --cpp_implementation not recognized
```

**Anything else we should know about your project / environment**

N/A

**Comment by adamjstewart** [comment]
Docs on the main branch have been updated to reflect that bazel is now required to build? https://github.com/protocolbuffers/protobuf/tree/main/python

**Comment by haberman** [comment]
Yes, the README has been updated to reflect the new state of the world.

The `--cpp_implementation` flag is obsolete and no longer supported.

You can either build from source packages (as published on PyPI), which will automatically build the upb implementation. Or you can build a source or binary package from the source repo using Bazel.
